### PR TITLE
MacOS Cross build on Linux.

### PR DESCRIPTION
**Pull request by dbalan**
Attempts to build the macOS binraries with clang cross compilation and osxcrossbuild project.

ToDo:
- [x] fix crashreporter

### CODE REVIEW

File: macdocker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,80 @@
+FROM ubuntu:17.10
```
[the remark this answers is not in the document]
Comment (dbalan) [reply macdocker/Dockerfile]: We already move the other docker file to its own repository while building, it makes sense move each into their own folders.

File: macdocker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+WORKDIR build
+RUN cmake ..
+RUN make
+
+# libdmg-hfsplus is broken on newer openssl api, but diskdev commands needs it.
+RUN apt-get update && \
+	apt-get install -y libssl-dev
+# diskdev
+WORKDIR /opt
+RUN curl -LO  http://cavan.codon.org.uk/~mjg59/diskdev_cmds/diskdev_cmds-540.1.linux3.tar.gz
```
[the remark this answers is not in the document]
Comment (dbalan) [reply macdocker/Dockerfile]: I believe mozilla gets this from their `tooltool` cache, I might be wrong.
Hmm, I don't think we actually need these tools for building (need it for debugging). If not I'll see if we can mirror it.

File: macdocker/get_dependencies.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+set -e -o pipefail
+
+if [[ -z $OSX_SDK_PATH ]]; then
+   OSX_SDK_PATH="s3://cliqz-ci/osxcross/MacOSX10.11.sdk.tar.bz2"
```
[the remark this answers is not in the document]
Comment (dbalan) [reply macdocker/get_dependencies.sh]: I don't know if the license allows re-distribution. Better to keep it private, one can easily generate it if they have access to a mac. This is the same reason this is not downloaded from the dockerfile.

File: mozilla-release/browser/config/cliqz-release.mozconfig [hunk 4]
```diff
@@ -4,8 +4,8 @@
 
 ac_add_options --enable-update-channel=${MOZ_UPDATE_CHANNEL}
 ac_add_options --enable-release
-ac_add_options --with-mozilla-api-keyfile=../mozilla-desktop-geoloc-api.key
-ac_add_options --with-google-api-keyfile=../google-desktop-api.key
+ac_add_options --with-mozilla-api-keyfile=$ROOT_PATH/mozilla-desktop-geoloc-api.key
```
[the remark this answers is not in the document]
Comment (dbalan) [reply mozilla-release/browser/config/cliqz-release.mozconfig]: good catch, I will enable back the linux builds and test how this goes.

File: osxcrossbuild.sh [hunk 5]
```diff
@@ -0,0 +1,115 @@
+#/usr/bin/env bash
```
[the remark this answers is not in the document]
Comment (dbalan) [reply osxcrossbuild.sh]: For this PR I didn't want to make magic more complicated than what it is right now, what do you think?

File: macdocker/Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (dbalan) [reply macdocker/Dockerfile]: We do need the `newfs_hfs` commands from it, but we might be able to replace it with `hformat` from hfstools. But for now, I suggest we use this.

**Comment by dbalan** [comment]
@chrmod The changes you requested are up. Can you have a look?

**Comment by alver-cliqz** [comment]
Closing this PR, current work with cross-building continue in osxcrossbuild branch (and will be add as separate PR after complete).

**Pull request closed without merging**